### PR TITLE
Fix for panic when `difference` is called with no data

### PR DESCRIPTION
**Pull request by adamperlin**
Address #393; Small bounds checking fix in `difference` implementation to prevent panic; added flux queries to demonstrate relevant behavior; updates to `difference` unit tests for cases when there are less than two rows.

### CODE REVIEW

File: query/functions/difference.go [hunk 1]
```diff
@@ -173,6 +173,10 @@ func (t *differenceTransformation) Process(id execute.DatasetID, b query.Block)
 		l := cr.Len()
 		for j, c := range cols {
 			d := differences[j]
+			if d == nil && l == 0 {
```
Comment (aanthony1243) [inline query/functions/difference.go]: should this be || ? 

or another approach,  if   `l == 0` we can just skip the loop completely, right?

Comment (adamperlin) [reply query/functions/difference.go]: whoops you're right - `l == 0` is all that's needed.

Comment (adamperlin) [reply query/functions/difference.go]: fixed

File: query/functions/difference.go [hunk 2]
```diff
@@ -173,7 +173,7 @@ func (t *differenceTransformation) Process(id execute.DatasetID, b query.Block)
 		l := cr.Len()
 		for j, c := range cols {
 			d := differences[j]
-			if d == nil && l == 0 {
+			if l == 0 {
```
Comment (aanthony1243) [inline query/functions/difference.go]: `l` is unchanged by this loop, so you're looping over cols for no reason.  
you probably want 
```
if l != 0 {
    for j, c := range cols {
     .....
   }
}
```

**Pull request merged**